dialer.go

```diff
@@ -34,6 +34,12 @@ func (df dialerFunctionWrapper) Call() (serverConn, string, error) {
 // value), and adding a new network type is as easy as writing the dialer
 // function and adding it to the map.
 func (w *Writer) getDialer() dialerFunctionWrapper {
+	if w.cdialer != nil {
+		// we use 'customDialer' as the name, since the custom dialer can technically
+		// override any network we pass in anyways
```

Comment:
Rather than making `network` meaningless, what if we added the requirement that if you want to use a custom dialer then you need to use the network `"custom"`. (This is how we added `"tcp+tls"` as an option for `network`, even though it also requires additional parameters just as this does.)

That would let us add this `dialerFunctionWrapper` to the `dialers` map below instead of creating this separate code path.

Reply:
Maybe I'm missing something, but if we say you have to use "custom" as the network, wouldn't that mean "custom" is passed in for every call to Dial? The idea was that they (the implementer of the dial function) could still use the real network string as they intended it (if they wanted to). But now that I look again, I guess changing the network/addr on the Writer isn't supported so those would never change anyways. Is that correct?

Reply:
I guess the question is, where do we want network to be meaningless? Because it will always be meaningless somewhere then. As I have it, network is still meaningful to the custom dialer, but not the getDialer/dialerFunctionWrapper. If we force "custom" it becomes meaningless to the custom dialer but special to getDialer/dialerFunctionWrapper.

Reply:
If my memory serves me correctly, the _only_ thing the `network` parameter does is specify which dialer to use. (If you can point me to a place I'm wrong about that, I'd appreciate it.)

But if I'm right, that means specifying a `network` _and_ a `customDialer` doesn't really make sense. You're not using that `network` to choose a dialer, you're specifying your own dialer.

Reply:
You're technically correct, in that it only matters to _this_ package when specifying the dialer. But every dialer passes it into the real dial function, and since the point of this change was to allow the user to add a custom dialer, they may want to use the network parameter. It just didn't feel right, even though I know this was how the other dialer methods worked. (Again, not specified in the godoc)

While I still disagree about this, I made the requested change anyways, and I did add a note about this one in the godoc, that you need to pass "custom". In the example I added I also mentioned they cannot simply pass the passed network as it won't work.

Reply:
Oddly, as with the other Dial functions, if you don't specify the correct network, they all act like the regular Dial. Meaning if you pass your own tls config for DialWithTLSConfig or dialer with DialWithCustomDialer your custom stuff gets ignored.